conans/model/conf.py

```diff
@@ -228,6 +235,8 @@ def get(self, conf_name, default=None, check_type=None):
                 return self._get_boolean_value(v)
             elif check_type is str and not isinstance(v, str):
                 return str(v)
+            elif v is None:  # value was unset
```

Comment:
🤔  when does this happen?

Reply:
Good point! This is happening when you unset a variable and you want to get its value (I added a few lines in `conans/test/unittests/model/test_conf.py` to test it):
```
[conf]
user.company.check:shared=!
```
```python
    assert c.get("user.company.check:shared") is None  # unset value
    assert c.get("user.company.check:shared", default=[]) == []  # returning default
    assert c.get("user.company.check:shared", default=[], check_type=list) == []  # not raising exception
```
Before my change, that `check_type=list` was raising an exception because the value was `None` but I thought it makes sense that you get the `default` value instead of raising the exception.